Namespace metrics under "app"

jason-o-matic opened the pull request:
Metrics as they existed before conflicted in some ways with
InstrumentalD. This clarifies that these metrics come from your
application code.

This also clarifies the service and exception namespaces, making them
more specific.

jason-o-matic committed:
Namespace metrics under "app"

Metrics as they existed before conflicted in some ways with
InstrumentalD. This clarifies that these metrics come from your
application code.

This also clarifies the service and exception namespaces, making them
more specific.

janxious commented on METRICS.MD:
```diff
@@ -5,21 +5,21 @@
 * `active_record.users.find`
```
There is a bunch of outdated information in here, and I think when the README links to the production metrician documentation, this file is probably redundant and can be deleted entirely. Thoughts?

janxious commented:
It occurred to me in a sleep vision that we could change none of the code in the gem and have a default prefix of "app".

`Metrician.prefix = "app"`

I think that works better than this solution in some ways, since in the world of this PR, if someone sets a prefix, they are going to get "<prefix>.app.", with no way to remove that leader.

janxious commented:
Also that could be set in the agent configuration instead of directly in metrician, which could be interesting.

jason-o-matic committed:
Default prefix

The previous version of this change had Metrician metrics prefixed with
"app", but you could add a custom prefix as well. This changes it so the
"app" prefix is set as a default, and you can change it using the prefix
setter on Metrician.

This is a little simpler, and allows the user to change the "app" prefix
if they want.

jason-o-matic committed:
Remove outdated metric list file

We link to the Instrumental documentation which covers this info and is
up to date.

jason-o-matic commented:
Updated if you want to gander again @janxious!

janxious commented on lib/metrician.rb:
```diff
@@ -53,12 +54,13 @@ def self.dotify(klass)
   # TODO: consider removal/movement to Instrumental Agent
   def self.prefix
-    @prefix || ""
+    @prefix || DEFAULT_PREFIX
```
Maybe this should be `||=` ?

janxious commented on lib/metrician.rb:
```diff
@@ -53,12 +54,13 @@ def self.dotify(klass)
 
   # TODO: consider removal/movement to Instrumental Agent
```
I think all the TODO comments around prefixes can be removed

janxious reviewed (approved):
I think there are some small changes to be made, but this is looking right.

jason-o-matic committed:
Remove TODOs that we decided not to do

Leaving prefixing here since Metrician can be thought of as its own
group of metrics which you may want to prefix as a group.

jason-o-matic committed:
Slight performance tweak
